plugins/dbus_announce.c

```diff
@@ -0,0 +1,144 @@
+#include "system.h"
+
+#include <dbus/dbus.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <rpm/rpmlog.h>
+#include <rpm/rpmts.h>
+#include <rpm/rpmdb.h>
+#include "lib/rpmplugin.h"
+
+struct dbus_announce_data {
+    DBusConnection * bus;
+    int old_dboffset;
+};
+
+static rpmRC dbus_announce_init(rpmPlugin plugin, rpmts ts)
+{
+    struct dbus_announce_data * state = rcalloc(1, sizeof(*state));
+    rpmPluginSetData(plugin, state);
+    return RPMRC_OK;
+}
+
+static void dbus_announce_close_bus(struct dbus_announce_data * state)
+{
+    if (state->bus) {
+	dbus_connection_close(state->bus);
+	dbus_connection_unref(state->bus);
+	state->bus = NULL;
+    }
+}
+
+static rpmRC open_dbus(rpmPlugin plugin, rpmts ts)
+{
+    DBusError err;
+    int rc = 0;
+    struct dbus_announce_data * state = rpmPluginGetData(plugin);
+
+    /* Already open */
+    if (state->bus)
+	return RPMRC_OK;
+
+    /* ...don't notify on test transactions */
+    if (rpmtsFlags(ts) & (RPMTRANS_FLAG_TEST|RPMTRANS_FLAG_BUILD_PROBS))
+	return RPMRC_OK;
+
+    /* ...don't notify on chroot transactions */
+    if (!rstreq(rpmtsRootDir(ts), "/"))
+	return RPMRC_OK;
+
+    dbus_error_init(&err);
+
+    state->bus = dbus_bus_get_private(DBUS_BUS_SYSTEM, &err);
+    if (dbus_error_is_set(&err)) {
+	goto err;
+    }
```

Comment:
There's no strict guideline whether to use {} on a one-liner block, but please pick a strategy and use it consistently. There are multiple cases where braces are not used (eg a few lines above), and many cases where they are not, throughout this file.

I personally tend to avoid the the excess {} unless it's needed for readability.

Reply:
Deleted lots of { and }.